tests/ci/integration/python_patch/main/aws-lc-cpython.patch

```diff
@@ -0,0 +1,555 @@
+ 
+-        with self.assertRaisesRegex(
+-                ssl.CertificateError,
+-                "IP address mismatch, certificate is not valid for "
+-                "'127.0.0.1'"):
++        regex = "IP address mismatch, certificate is not valid for '127.0.0.1'"
++        if ssl is not None and "AWS-LC" in ssl.OPENSSL_VERSION:
++            regex = "CERTIFICATE_VERIFY_FAILED"
++        with self.assertRaisesRegex(ssl.CertificateError, regex):
+             with self.reaped_server(SimpleIMAPHandler) as server:
+                 client = self.imap_class(*server.server_address,
+                                          ssl_context=ssl_context)
+diff --git a/Lib/test/test_site.py b/Lib/test/test_site.py
+index 0502181854..e23531f147 100644
+--- a/Lib/test/test_site.py
++++ b/Lib/test/test_site.py
+@@ -564,6 +564,7 @@ def test_customization_modules_on_startup(self):
+     def test_license_exists_at_url(self):
+         # This test is a bit fragile since it depends on the format of the
+         # string displayed by license in the absence of a LICENSE file.
++        #import ssl
+         url = license._Printer__data.split()[1]
+         req = urllib.request.Request(url, method='HEAD')
+         # Reset global urllib.request._opener
+diff --git a/Lib/test/test_ssl.py b/Lib/test/test_ssl.py
+index 3fdfa29605..6f6fea7c36 100644
+--- a/Lib/test/test_ssl.py
++++ b/Lib/test/test_ssl.py
+@@ -41,6 +41,7 @@
+ from ssl import TLSVersion, _TLSContentType, _TLSMessageType, _TLSAlertType
+ 
+ Py_DEBUG_WIN32 = support.Py_DEBUG and sys.platform == 'win32'
++Py_OPENSSL_IS_AWSLC = "AWS-LC" in ssl.OPENSSL_VERSION
+ 
+ PROTOCOLS = sorted(ssl._PROTOCOL_NAMES)
+ HOST = socket_helper.HOST
+@@ -167,7 +168,7 @@ def is_ubuntu():
+     except FileNotFoundError:
+         return False
+ 
+-if is_ubuntu():
++if is_ubuntu() and not Py_OPENSSL_IS_AWSLC:
+     def seclevel_workaround(*ctxs):
+         """"Lower security level to '1' and allow all ciphers for TLS 1.0/1"""
+         for ctx in ctxs:
+@@ -544,7 +545,7 @@ def test_openssl_version(self):
+         else:
+             openssl_ver = f"OpenSSL {major:d}.{minor:d}.{fix:d}"
+         self.assertTrue(
+-            s.startswith((openssl_ver, libressl_ver)),
++            s.startswith((openssl_ver, libressl_ver, "AWS-LC")),
+             (s, t, hex(n))
+         )
+ 
+@@ -1102,7 +1103,6 @@ def test_min_max_version(self):
+ 
+         if has_tls_protocol(ssl.PROTOCOL_TLSv1_1):
+             ctx = ssl.SSLContext(ssl.PROTOCOL_TLSv1_1)
+-
+             self.assertIn(
+                 ctx.minimum_version, minimum_range
+             )
+@@ -1162,24 +1162,25 @@ def test_load_cert_chain(self):
+         with self.assertRaises(OSError) as cm:
+             ctx.load_cert_chain(NONEXISTINGCERT)
+         self.assertEqual(cm.exception.errno, errno.ENOENT)
+-        with self.assertRaisesRegex(ssl.SSLError, "PEM lib"):
++        with self.assertRaisesRegex(ssl.SSLError, "PEM (lib|routines)"):
```

Comment:
Should we standardize our error strings?

Reply:
That's certainly an option. If we don't consider error string contents part of our backwards compatibility "contract", then we're free to change them to hew more closely to some version(s) of OpenSSL. The scope of such an effort would probably be pretty sizeable, and I don't know if it would provide any material value other than satisfying brittle tests that key on error string contents.